src/wordlift/mappings/class-mappings-dbo.php

```diff
@@ -271,20 +295,30 @@ public function get_rule_groups_by_mapping( $mapping_id ) {
 	/**
 	 * Gets a list of rule items belong to rule_group_id.
 	 *
-	 * @param int $rule_group_id Indicates which group the item belongs to.
+	 * @param string $rule_group_id Indicates which group the item belongs to.
 	 *
 	 * @return array Get list of rule items.
 	 */
 	public function get_rules_by_rule_group( $rule_group_id ) {
-		$rule_table_name = $this->wpdb->prefix . WL_RULE_TABLE_NAME;
+		$rule_table_name  = $this->wpdb->prefix . WL_RULE_TABLE_NAME;
+		$rule_group_id    = (string) $rule_group_id;
+		$ui_defined_rules = array();
+		if ( is_numeric( $rule_group_id ) ) {
+			$ui_defined_rules = $this->wpdb->get_results(
+				$this->wpdb->prepare(
+					"SELECT * FROM $rule_table_name WHERE rule_group_id=%d",
+					(int) $rule_group_id
+				),
+				ARRAY_A
+			);
+		}
+		$manual_rules = apply_filters( 'wl_mappings_register_rules', array() );
```

Comment:
@naveen17797 pls

- [ ] add PhpDoc for the filter

- [ ] change filter name to `wl_mappings_rules`

- [ ] provide `$ui_defined_rules` as initial value instead of `array()` so that hooks can modify existing data, and remove `array_merge` at ln 321.

- [ ] add `$rule_group_id` as parameter and remove `array_filter` at ln 317.